sycl/test-e2e/Graph/nodes_profiling_info.cpp

```diff
@@ -0,0 +1,331 @@
+#if GRAPH_TESTS_VERBOSE_PRINT
+    auto endKernelSubmit1 = std::chrono::high_resolution_clock::now();
+    auto StartKernelSubmit2 = std::chrono::high_resolution_clock::now();
+#endif
+    KernelEvent2 = Queue.submit(
+        [&](handler &CGH) { CGH.ext_oneapi_graph(KernelGraphExec); });
+    Queue.wait_and_throw();
+#if GRAPH_TESTS_VERBOSE_PRINT
+    auto endKernelSubmit2 = std::chrono::high_resolution_clock::now();
+
+    double DelayCopy = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                           EndCopyGraph - StartCopyGraph)
+                           .count();
+    std::cout << "Copy Graph delay (in ns) : " << DelayCopy << std::endl;
+    double DelayKernel1 = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                              endKernelSubmit1 - StartKernelSubmit1)
+                              .count();
+    std::cout << "Kernel 1st Execution delay (in ns) : " << DelayKernel1
+              << std::endl;
+    double DelayKernel2 = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                              endKernelSubmit2 - StartKernelSubmit2)
+                              .count();
+    std::cout << "Kernel 2nd Execution delay (in ns) : " << DelayKernel2
+              << std::endl;
+#endif
+
+    // The copy graph is only made of a single node so all graph timestamps must
+    // be the same that node timestamps.
+    auto CopySubmit =
+        CopyEvent
+            .get_profiling_info<sycl::info::event_profiling::command_submit>();
+    auto CopyStart =
+        CopyEvent
+            .get_profiling_info<sycl::info::event_profiling::command_start>();
+    auto CopyEnd =
+        CopyEvent
+            .get_profiling_info<sycl::info::event_profiling::command_end>();
+
+    assert(verifyProfiling(CopyEvent, NodeCopy));
+    auto NodeCopySubmit = CopyEvent.ext_oneapi_get_profiling_info<
+        sycl::info::event_profiling::command_submit>(NodeCopy);
+    auto NodeCopyStart = CopyEvent.ext_oneapi_get_profiling_info<
+        sycl::info::event_profiling::command_start>(NodeCopy);
+    auto NodeCopyEnd = CopyEvent.ext_oneapi_get_profiling_info<
+        sycl::info::event_profiling::command_end>(NodeCopy);
+
+    assert((CopySubmit == NodeCopySubmit) && "Copy submit times differ");
+    assert((CopyStart == NodeCopyStart) && "Copy start times differ");
+    assert((CopyEnd == NodeCopyEnd) && "Copy end times differ");
+
+    // Check first execution
+
+    auto Submit =
+        KernelEvent1
+            .get_profiling_info<sycl::info::event_profiling::command_submit>();
+    auto Start =
+        KernelEvent1
+            .get_profiling_info<sycl::info::event_profiling::command_start>();
+    auto End =
+        KernelEvent1
+            .get_profiling_info<sycl::info::event_profiling::command_end>();
+
+    std::vector<uint64_t> NodeSubmit(Nodes.size());
+    std::vector<uint64_t> NodeStart(Nodes.size());
+    std::vector<uint64_t> NodeEnd(Nodes.size());
+    uint64_t GlobalRuntime;
```

Comment:
Should initialize this since we're loading from it with `+=` 
```suggestion
    uint64_t GlobalRuntime = 0;
```